files-csharp/CountFilesInaFolderUsingCSharp/CountFilesInaFolderTests/FileCounterUnitTests.cs

```diff
@@ -0,0 +1,51 @@
+namespace CountFilesInaFolderTests;
+
+public class FileCounterUnitTests : IDisposable
+{
+    private readonly string _tempDirectory;
+    private readonly int _expectedFileCount = 1000;
+
+    public FileCounterUnitTests()
+    {
+        _tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        Directory.CreateDirectory(_tempDirectory);
+
+        for (var i = 0; i < 1000; i++)
+        {
+            File.Create(Path.Combine(_tempDirectory, $"file{i}.txt")).Dispose();
+        }
+    }
```

Comment:
This along with the Dispose should be moved into a ClassFixture so that we only generate the files once and cleanup once. As currently written we are generating the 1000 files on each test execution.